test/unit_test/rag/svr/test_task_executor_terminal_log.py

```
@@ -0,0 +1,51 @@
+    helper = next(
+        node
+        for node in tree.body
+        if isinstance(node, ast.FunctionDef) and node.name == "_record_terminal_pipeline_log_best_effort"
+    )
+    namespace = {
+        "PipelineOperationLogService": SimpleNamespace(create=lambda **_kwargs: "saved"),
+        "PipelineTaskType": SimpleNamespace(PARSE="parse"),
+        "get_recording_context": lambda: SimpleNamespace(save_func_return_value=lambda *_args: None),
+        "logging": logging,
+    }
+    exec(compile(ast.Module(body=[helper], type_ignores=[]), "task_executor.py", "exec"), namespace)
```

Ruff (0.16.2)

[error] Use of `exec` detected

(S102)